api/v1beta1/cluster_types.go

```diff
@@ -669,17 +672,20 @@ type MachineDeploymentTopology struct {
 	// class is the name of the MachineDeploymentClass used to create the set of worker nodes.
 	// This should match one of the deployment classes defined in the ClusterClass object
 	// mentioned in the `Cluster.Spec.Class` field.
+	// +kubebuilder:validation:MaxLength=256
```

Comment:
253 I think, name of a custom resource has a limit of 253 chars

Given this is required, should also set a minimum

Reply:
This is not referring to a CRD. It is referring to a MachineDeploymentClass in the ClusterClass (which also has MaxLength 256 for its name now)

Reply:
Ahh, I thought that was a CR reference, ignore me